import_customer/models/import_customer.py

```diff
@@ -0,0 +1,86 @@
+import base64
+from lxml import etree as ET
+from odoo.exceptions import ValidationError
+from odoo import _, api, fields, models
+
+
+class ImportCustomer(models.Model):
+    _name = "import.customer"
+    _description = "Import Customer"
+    _rec_name = "name"
+
+    # Added new fields #T7156
+    partner_ids = fields.Many2many(
+        "res.partner",
+        string="Partner",
+    )
+    created_partner = fields.Integer(
+        compute="_compute_created_partner", string="Customers"
+    )
+    name = fields.Char(default=lambda self: ("New"), string="Number")
+    partner_name = fields.Char(string="Name")
+    import_file = fields.Binary()
+    import_file_name = fields.Char(string="File")
+
+    @api.model_create_multi
+    def create(self, vals_ls):
+        """Inherited method to assign a sequence on create #T7156"""
+        for vals in vals_ls:
+            vals["name"] = self.env["ir.sequence"].next_by_code("import.customer")
+            return super().create(vals)
+
+    def import_customer(self):
+        """New method will parse the uploaded xml file and create a new res_partner
+        record based on it #T7156"""
+        if not self.import_file:
+            raise ValidationError(_("Please upload a file to import customers."))
+        file_as_str = base64.b64decode(self.import_file)
+        try:
+            xml = ET.fromstring(file_as_str)
+        # if the XML is missing root tags raise error. #T7156
+        except ET.XMLSyntaxError:
+            raise ValidationError(_("The uploded file is missing an XML tag."))
+        customer_ls = [
+            {child.tag: child.text for child in customer} for customer in xml
+        ]
+        # creating new res_partner records and storing their ids for
+        # the smart button. #T7156
+        partner_ids = [
+            self.env["res.partner"]
+            .create(
+                {
+                    "name": customer.get("CustomerName"),
```

Comment:
- Add new m2o of import_customer at the partner level for reference.
- Also set import_customer records reference in Partners.